server/master/src/main/java/org/apache/accumulo/master/tableOps/MoveExportedFiles.java

```diff
@@ -48,21 +55,35 @@
 
       Map<String,String> fileNameMappings = PopulateMetadataTable.readMappingFile(fs, tableInfo);
 
-      for (String oldFileName : fileNameMappings.keySet()) {
-        if (!fs.exists(new Path(tableInfo.exportDir, oldFileName))) {
-          throw new AcceptableThriftTableOperationException(tableInfo.tableId, tableInfo.tableName,
-              TableOperation.IMPORT, TableOperationExceptionType.OTHER,
-              "File referenced by exported table does not exists " + oldFileName);
-        }
-      }
+      FileStatus[] exportedFiles = fs.listStatus(new Path(tableInfo.exportDir));
+      FileStatus[] importedFiles = fs.listStatus(new Path(tableInfo.importDir));
+
+      Set<String> importing = Arrays.stream(exportedFiles).map(fstat -> fstat.getPath().getName())
+        .map(fileNameMappings::get).collect(Collectors.toSet());
 
-      FileStatus[] files = fs.listStatus(new Path(tableInfo.exportDir));
+      Set<String> imported = Arrays.stream(importedFiles).map(fstat -> fstat.getPath().getName())
+        .collect(Collectors.toSet());
+      log.debug("Files already present in imported (target) directory: {}",
+        StringUtils.join(imported.toArray(), ","));
```

Comment:
In addition to Keith's suggestion, we'll need to be careful about the use of commons-lang here when we merge to the master branch, which now uses commons-lang3. If you wanted to avoid that, you could instead use the built-in Java Joiner, as in:

```suggestion
        imported.stream().collect(Collectors.joining(",")));
```

It's only slightly longer, but removes an unneeded library dependency.

Reply:
Addressed in latest commit